v-2.0/producer/src/main/java/io/sweetwater/developer/producer/ProducerCloudApplication.java

```diff
@@ -0,0 +1,53 @@
+package io.sweetwater.developer.producer;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+import java.time.Instant;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.context.event.EventListener;
+import org.springframework.boot.context.event.ApplicationStartedEvent;
+
+import io.sweetwater.developer.avro.ElectronicOrder;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import lombok.RequiredArgsConstructor;
+
+/**
+ProducerCloudApplication class uses Spring boot to create a Producer.
+Uses KafkaTemplate to produce message to global source topic.
+ */
+@SpringBootApplication
+public class ProducerCloudApplication {
+	public static void main(String[] args) {
+		SpringApplication.run(ProducerCloudApplication.class, args);
+	}
+}
+
+@RequiredArgsConstructor
+@Component
+class Producer {
+    private final KafkaTemplate<String, ElectronicOrder> kafkaTemplate;
```

Comment:
is this kafkareplying template?

Reply:
ReplyingKafkaTemplate extends KafkaTempalte. https://docs.spring.io/spring-kafka/api/org/springframework/kafka/requestreply/ReplyingKafkaTemplate.html
Will plan to change in coming minor version.